File: chor-extraction/src/main/java/ast/sp/nodes/Condition.java
package ast.sp.nodes;

import ast.sp.interfaces.Behaviour;

/**
 * Created by lara on 04/04/17.
 */
public class Condition implements Behaviour{
    private final String process;
    private final String expression;
    private final Behaviour thenBehaviour;
    private final Behaviour elseBehaviour;

    public Condition(String process, String expression, Behaviour thenBehaviour, Behaviour elseBehaviour) {
        this.process = process;
        this.expression = expression;
        this.thenBehaviour = thenBehaviour;
        this.elseBehaviour = elseBehaviour;
    }

    public String getProcess() {
        return process;
    }

    public String getExpression() {
        return expression;
    }

    public Behaviour getThenBehaviour() {
        return thenBehaviour;
    }

    public Behaviour getElseBehaviour() {
        return elseBehaviour;
    }

    public String toString() {
        return "if " + process + "." + expression + " then " + thenBehaviour.toString() + " else " + elseBehaviour.toString();
    }
}

File: pe/src/main/kotlin/ast/cc/nodes/Condition.kt
package ast.cc.nodes

import ast.cc.interfaces.CCVisitor
import ast.cc.interfaces.ChoreographyBody

data class Condition(val process: String, val expression: String, val thenChoreography: ChoreographyBody, val elseChoreography: ChoreographyBody) : ChoreographyBody() {
    override fun <T> accept(visitor: CCVisitor<T>): T = visitor.visit(this)

    override fun toString(): String = "if $process.$expression then $thenChoreography else $elseChoreography"
}
